source/common/stats/thread_local_store.cc

```diff
@@ -34,6 +34,48 @@ ThreadLocalStoreImpl::~ThreadLocalStoreImpl() {
   ASSERT(scopes_.empty());
 }
 
+void ThreadLocalStoreImpl::setStatsMatcher(StatsMatcherPtr&& stats_matcher) {
```

Comment:
This seems unrelated to filesystems. Thoughts on moving this part of the change to a different PR?

Reply:
The reason I needed to tackle this was that pre-declaring the stats for the filesystem at API creation time made them all leak through the late application of the StatsMatcher and broke the StatsMatcher integration test, which already had one hardcoded exception in it.

I felt it better to clean up the matcher than had a lot more exceptions to the integration-test.

However what I can do is put that into its own PR which we can review first.

Reply:
Great, SGTM. Thanks for breaking things apart.

Reply:
Pulled into #5015 which can be reviewed without the file-system change.

Reply:
#5105 is merged now and so is no longer part of this PR.